src/core/Akka/Configuration/Hocon/HoconObject.cs

```diff
@@ -196,6 +196,8 @@ public void Merge(HoconObject other)
                     //if both values are objects, merge them
                     if (thisItem.IsObject() && otherItem.Value.IsObject())
                         thisItem.GetObject().Merge(otherItem.Value.GetObject());
+                    else
```

Comment:
ok, so if I'm reading this right, if both items are objects, they are to be merged, but how is the alternative supposed to work? what is otherItem's type?

Reply:
HOCON have 3 basic datatypes, `Object`, `Array`, and `Literal`. 
`Array` and `Literal` will always override merged object fields, but when an `Object` met another `Object`, they got merged instead.

Reply:
The full specification for HOCON can be read here: https://github.com/akkadotnet/hocon

Reply:
ah ok, I see it now

Reply:
let's give it a shot then